cpp/src/strings/repeat_strings.cu

```diff
@@ -172,22 +199,259 @@ std::unique_ptr<column> repeat_strings(strings_column_view const& input,
                              mr);
 }
 
+namespace {
+constexpr auto invalid_size = std::numeric_limits<size_type>::lowest();
+constexpr auto max_offset   = static_cast<int64_t>(std::numeric_limits<size_type>::max());
+
+template <typename Func>
+std::tuple<std::unique_ptr<column>, size_type, rmm::device_buffer, size_type>
+generate_offsets_and_null_mask(Func& fn,
+                               size_type strings_count,
+                               bool generate_null_mask,
+                               check_output_overflow overflow_checking,
+                               rmm::cuda_stream_view stream,
+                               rmm::mr::device_memory_resource* mr)
+{
+  auto offsets_column = make_numeric_column(
+    data_type{type_id::INT32}, strings_count + 1, mask_state::UNALLOCATED, stream, mr);
+  auto offsets_view = offsets_column->mutable_view();
+  auto d_offsets    = offsets_view.template data<int32_t>();
+  fn.d_offsets      = d_offsets;
+
+  auto validities = rmm::device_uvector<int8_t>(generate_null_mask ? strings_count : 0, stream);
+  fn.d_validities = generate_null_mask ? validities.begin() : nullptr;
+
+  // Compute sizes of the output strings (and null mask, if needed).
+  thrust::for_each_n(
+    rmm::exec_policy(stream), thrust::make_counting_iterator<size_type>(0), strings_count, fn);
+
+  // Compute the offsets values.
+  if (overflow_checking == check_output_overflow::YES) {
+    CUDA_TRY(cudaMemsetAsync(d_offsets + strings_count, 0, sizeof(offset_type), stream.value()));
+    thrust::exclusive_scan(rmm::exec_policy(stream),
+                           d_offsets,
+                           d_offsets + strings_count + 1,
+                           d_offsets,
+                           size_type{0},
+                           [] __device__(auto const lhs, auto const rhs) {
+                             if (lhs == invalid_size || rhs == invalid_size) {
+                               return invalid_size;
+                             }
+                             auto const sum = static_cast<int64_t>(lhs) + static_cast<int64_t>(rhs);
+                             return sum > max_offset ? invalid_size : static_cast<size_type>(sum);
+                           });
```

Comment:
I think you could just do a `transform_reduce` on the sizes to get the total size. Then you would not need to place these invalid sizes in the output.

Reply:
My idea is that, if some of the input strings are very large (like strings having size as `[INT_MAX/2, 10, 15,......]`). If we repeat all the strings 100 times, size of the first string will exceed `INT_MAX`. In such case, I want to return something which can tell the caller immediately that there will be overflow. I'm still not sure how to tell in a more exact way: overflow due to individual string, or overflow due to prefix sum of the offsets.